openedx-unsupported/ecommerce
# LEARNER-8973 Add original_transaction_id

# ⛔️ DEPRECATION WARNING

**This repository is deprecated and in maintainence-only operation while we work on a replacement, please see [this announcement](https://discuss.openedx.org/t/deprecation-removal-ecommerce-service-depr-22/6839) for more information.**

Although we have stopped integrating new contributions, we always appreciate security disclosures and patches sent to [security@edx.org](mailto:security@edx.org)



## Anyone internally merging to this repository is expected to [release and monitor their changes](https://openedx.atlassian.net/wiki/spaces/RS/pages/1835106870/How+to+contribute+to+our+repositories); if you are not able to do this DO NOT MERGE, please coordinate with someone who can to ensure that the changes are released.

## Required Testing
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)

## Description

Added a field original_transaction_id to be recorded in case of payment via iOS app.
This field is needed to identify redundant purchases in iOS and deny access to course in case a redundant purchase request has been made.

## Supporting information

- Link to Jira: https://2u-internal.atlassian.net/browse/LEARNER-8973
- Link to Apple documentation for field original_transaction_id: https://developer.apple.com/documentation/appstorereceipts/original_transaction_id

## Testing instructions

To be tested via iOS mobile app. Testing steps include:

- Sign in on iOS app with userA.
- Purchase a course from userA.
- Sign out from edx app from userA and sign in to userB on same device.
- Attempt purchase again for the same course
- Access to course should be denied.
